.github/workflows/pr-checks.yml

```diff
@@ -72,9 +74,18 @@ jobs:
       - name: Validate Schema
         run: make test
 
-      # Ensure there is no diff when make gen-schema-json is run 
-      - run: make gen-schema-json
-      - name: Check no diff 
+      # Sync JSON schemas from YAML versions
+      - name: Sync JSON from YAML
         run: |
-          if [ ! -z "$(git status --porcelain)" ]; then echo "JSON schema generation produced diff. Run 'make gen-schema-json' and commit results."; exit 1; fi
+          make gen-schema-json
+          if [ ! -z "$(git status --porcelain)" ]; then
+            echo "JSON schema generation produced diff, regenerating and committing result...";
+            git config --global user.name "validate-and-sync-schema-job"
+            git config --global user.email "bot@openfeature.dev"
+            git add json/*.json
+            git commit -m 'chore: sync json schema from yaml' -s
+            git push
+          else
+            echo "No delta between YAML and JSON schemas, finishing..."
+          fi
```

Comment:
If there is a diff, this regenerates the JSON from the YAML, and commits is back - this way when release please updates the YAML files based on the `x-release-please` comment, it regenerates the JSON files to match. I tested it out [here](https://github.com/open-feature/flagd-schemas/pull/162/files#diff-bf4d4b61995cab1ef8258e33f5eb9fabbd274362bc1f0b147c0c64860d2bd6f0).